Add additional message when backendfs is extfs without d_type support

cgxu519 opened the pull request:
Signed-off-by: Chengguang Xu <cgxu519@gmx.com>

Attempt to provide more meaningful warning message.

GordonTheTurtle commented:
<!-- AUTOMATED:POULE:DCO-EXPLANATION -->
Please sign your commits following these rules:
https://github.com/moby/moby/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "overlayutils-addmsg" git@github.com:cgxu519/moby.git somewhere
$ cd somewhere
$ git rebase -i HEAD~842353651248
editor opens
change each 'pick' to 'edit'
save the file and quit
$ git commit --amend -s --no-edit
$ git rebase --continue # and repeat the amend for each commit
$ git push -f
~~~

Amending updates the existing PR. You **DO NOT** need to open a new one.

Commit:
Add additional message when backendfs is extfs without d_type support

ext4 support d_type by default, but filetype feature is a tunable so
there is still a chance to disable it for some reasons. In this case,
print additional message to explicitly tell how to support d_type.

Signed-off-by: Chengguang Xu <cgxu519@gmx.com>

thaJeztah commented:
ping @kolyshkin PTAL

cgxu519 commented:
ping @kolyshkin

thaJeztah reviewed (approved):
LGTM

kolyshkin commented on daemon/graphdriver/overlayutils/overlayutils.go:
```diff
@@ -14,6 +14,11 @@ func ErrDTypeNotSupported(driver, backingFs string) error {
 	}
+
+	if backingFs == "extfs" {
+		msg += " Reformat the filesystem with -O filetype to enable d_type support."
```
Maybe even
```golang
msg += " Reformat the filesystem (or use tune2fs) with -O filetype flag to enable d_type support."
```

cgxu519 replied:
Thanks for your review, I've added more information about tune2fs.

thaJeztah reviewed (approved):
still LGTM, thanks!

thaJeztah commented:
I see @kolyshkin's comments are addressed; I'll go ahead and merge

thaJeztah commented:
Thanks for you contribution @cgxu519 !
